server/jetstream_api.go

```diff
@@ -1012,30 +1012,48 @@ func (s *Server) sendAPIErrResponse(ci *ClientInfo, acc *Account, subject, reply
 
 const errRespDelay = 500 * time.Millisecond
 
-func (s *Server) sendDelayedAPIErrResponse(ci *ClientInfo, acc *Account, subject, reply, request, response string, rg *raftGroup) {
-	js := s.getJetStream()
-	if js == nil {
-		return
-	}
-	var quitCh <-chan struct{}
-	js.mu.RLock()
-	if rg != nil && rg.node != nil {
-		quitCh = rg.node.QuitC()
-	}
-	js.mu.RUnlock()
+type delayedAPIResponse struct {
+	ci       *ClientInfo
+	acc      *Account
+	subject  string
+	reply    string
+	request  string
+	response string
+	rg       *raftGroup
+	after    time.Time
+}
 
-	s.startGoRoutine(func() {
-		defer s.grWG.Done()
+func (s *Server) delayedAPIResponder() {
+	defer s.grWG.Done()
+	for {
+		resp, ok := s.delayedAPIResponses.popOne()
+		if !ok {
+			return // IPQueue was shut down.
```

Comment:
No, this means that the queue is empty, which if you respond to a notification that there is something should not happen, expect in cases with drain(). Since I don't see the select on the queue's `ch`, this function should have returned right away...